barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -516,27 +516,39 @@ void vk_as_fields(const std::string& vk_path, const std::string& output_path)
  *
  * @param bytecode_path Path to the file containing the serialised bytecode
  * @param calldata_path Path to the file containing the serialised calldata (could be empty)
- * @param crs_path Path to the file containing the CRS (ignition is suitable for now)
+ * @param public_inputs_path Path to the file containing the serialised avm public inputs
+ * @param hints_path Path to the file containing the serialised avm circuit hints
  * @param output_path Path (directory) to write the output proof and verification keys
  */
 void avm_prove(const std::filesystem::path& bytecode_path,
                const std::filesystem::path& calldata_path,
+               const std::filesystem::path& public_inputs_path,
+               const std::filesystem::path& hints_path,
                const std::filesystem::path& output_path)
 {
     // Get Bytecode
     std::vector<uint8_t> const avm_bytecode =
         bytecode_path.extension() == ".gz" ? gunzip(bytecode_path) : read_file(bytecode_path);
-    std::vector<uint8_t> call_data_bytes{};
+
+    std::vector<uint8_t> calldata_bytes{};
     if (std::filesystem::exists(calldata_path)) {
-        call_data_bytes = read_file(calldata_path);
+        calldata_bytes = read_file(calldata_path);
+    }
+    std::vector<fr> const calldata = many_from_buffer<fr>(calldata_bytes);
+
+    std::vector<uint8_t> const avm_public_inputs = read_file(public_inputs_path);
+    std::vector<uint8_t> avm_hints;
+    try {
+        avm_hints = read_file(hints_path);
```

Comment:
Feels like this PR should include the deserialization of the hints on the CPP side, to be sure that the serde works.